thorlcr/thorutil/thormisc.hpp

```diff
@@ -306,22 +306,62 @@ class graph_decl CTimeoutTrigger : public CInterface, implements IThreaded
     virtual bool action() = 0;
 };
 
+// Tracks the current and peak storage used for some files
+class CFileSizeTracker: public CInterface
+{
+    RelaxedAtomic<unsigned __int64> activeSize{0};
```

Comment:
minor: but noteSpill is now an offset_t (as per IFile::size() and CFileSizeTracker is dealing with file sizes. I think these and all methods below should really also be changed to offset_t's